### ISSUE TITLE
Is configuration of colums/rows available in XLSX format yet?

### ISSUE THREAD

**Issue by sumeshk** [agershun/alasql#574]
Hi,

This is not an issue. 
Is the option of providing configuration values available in XLSX yet? I read in another post in dated mid 2015 that it was being worked on.

Is it available yet?

I would like to use custom config similar to "mystyle" similar to the below line.

`alasql('SELECT * INTO XLS("john.xls",?) FROM ?',[mystyle,$scope.items]);`

This was taken from the example : http://jsfiddle.net/agershun/95j0txwx/2/

Thanks,
Sumesh

**Comment by sumeshk** [comment]
Also, I wanted to add that I would have been okay with xls had it not been for the warning
"The file format and extension of 'john.xls' don't match. The file could be corrupted or unsafe. Unless ...". Editing the registry would not help me as multiple customers will be using this.

**Comment by mathiasrw** [comment]
@sumeshk yes - the warning is really poopy. we are not the only ones with this problem. Maybe @agershun kan tell if we can give XLSX configuration variables now?

**Comment by agershun** [comment]
I have idea to generate true XLSX format (that is why we need to include jszip inside the library). 

Could you try XLSXML() output function (it works similar way as XLSX() but without ```xlsx.js```)?

The only problem to finish this work is time... :(

**Comment by sumeshk** [comment]
XLSXML exports to .xml, isnt it? My client won't agree to any other format other than xls/xlsx.

**Comment by agershun** [comment]
If you name it *.xlsx, Excel will read it as XLSX.
```
SELECT * INTO XLSXML("myfile.xlsx") FROM ?
```
Or *.xls - I can not check it right now.

2016-02-01 16:49 GMT+03:00 sumeshk <notifications@github.com>:

**Comment by sumeshk** [comment]
`SELECT * INTO XLSXML("myfile.xlsx") FROM ?`
wont work.

`SELECT * INTO XLSXML("myfile.xls") FROM ?`

works. But it works exactly the same way it is with 
`SELECT * INTO XLS("myfile.xls") FROM ?`. Still showing the warning.

**Comment by agershun** [comment]
So, we need to generate zipped xlsx file. It is possible, but requires
about a week to finish it (also +5K for jszip..).

2016-02-01 17:09 GMT+03:00 sumeshk <notifications@github.com>:

**Comment by alvindotnetdeveloper** [comment]
@agershun and @mathiasrw  First of all thank you so much for this library. Simple and easy to use. I was wondering at this point of time does library support styles when exporting with **.xlsx** extension?

**Comment by mathiasrw** [comment]
Hi @aallvviinn. Thank you for the kind words. 

> does library support styles when exporting with .xlsx extension?

Nope, at the moment it does not. If you find a way or a library that supports styling a .xlsx please let us know.

**Comment by aallvviinn** [comment]
@mathiasrw I was exploring multiple libraries and found exceljs does support styling with a .xlsx extension. Hope it helps.

**Comment by mathiasrw** [comment]
Hmmm. Interesting.
